File: demos/examples/app/src/main/java/checkout/com/demo/MainActivity.java
package checkout.com.demo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import checkout.com.demo.Demos.CustomFieldsDemo;
import checkout.com.demo.Demos.CustomisationDemo;
import checkout.com.demo.Demos.HeadlessDemo;

public class MainActivity extends Activity {

    private Button mCustomisationDemo;
    private Button mCustomFieldsDemo;
    private Button mHeadlessDemo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mCustomisationDemo = findViewById(R.id.customisation_demo);
        mCustomFieldsDemo = findViewById(R.id.custom_fields_demo);
        mHeadlessDemo = findViewById(R.id.headless_demo);

        mCustomisationDemo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(MainActivity.this, CustomisationDemo.class);
                MainActivity.this.startActivity(myIntent);
            }
        });

        mCustomFieldsDemo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(MainActivity.this, CustomFieldsDemo.class);
                MainActivity.this.startActivity(myIntent);
            }
        });

        mHeadlessDemo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(MainActivity.this, HeadlessDemo.class);
                MainActivity.this.startActivity(myIntent);
            }
        });

    }
}

File: tokengenerator/src/main/java/com/caracode/tokengenerator/MainActivity.kt
package com.caracode.tokengenerator

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View.GONE
import android.view.View.VISIBLE
import com.checkout.sdk.CheckoutClient
import com.checkout.sdk.core.RequestMaker
import com.checkout.sdk.core.TokenResult
import com.checkout.sdk.request.CardTokenizationRequest
import com.checkout.sdk.utils.Environment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val tokenCallback = createTokenCallback()
        val checkoutClient = createCheckoutClient(tokenCallback)
        val requestMaker = createRequestMaker(checkoutClient)
        generate_token_button.setOnClickListener {
            progress_bar.visibility = VISIBLE
            requestMaker.makeTokenRequest(createCardTokenizationRequest())
        }
    }

    private fun createTokenCallback(): CheckoutClient.TokenCallback {
       return CheckoutClient.TokenCallback { tokenResult ->
            when (tokenResult) {
                is TokenResult.TokenResultSuccess -> setSuccessText(tokenResult)
                is TokenResult.TokenResultTokenisationFail -> setTokenizationFail(tokenResult)
                is TokenResult.TokenResultNetworkError -> setNetworkFail(tokenResult)
            }
            progress_bar.visibility = GONE
        }
    }

    private fun createRequestMaker(checkoutClient: CheckoutClient): RequestMaker {
        return RequestMaker.create(this, checkoutClient)
    }

    private fun createCheckoutClient(tokenCallback: CheckoutClient.TokenCallback): CheckoutClient {
        return CheckoutClient(this, KEY, Environment.SANDBOX, tokenCallback)
    }

    fun createCardTokenizationRequest(): CardTokenizationRequest {
        return CardTokenizationRequest(
            "4242424242424242",
            "Jim Stynes",
            "06",
            "2020",
            "100",
            null
        )
    }

    private fun setSuccessText(tokenSuccess: TokenResult.TokenResultSuccess) {
        token_text_view.text = getString(R.string.token_success_format, tokenSuccess.response.id)
    }

    private fun setTokenizationFail(tokenizationFail: TokenResult.TokenResultTokenisationFail) {
        token_text_view.text =
            getString(R.string.token_failed_tokenization_format, tokenizationFail.error.errorCode)
    }

    private fun setNetworkFail(networkFail: TokenResult.TokenResultNetworkError) {
        token_text_view.text =
            getString(R.string.token_failed_tokenization_format, networkFail.exception)
    }

    companion object {
        const val KEY = "pk_test_6e40a700-d563-43cd-89d0-f9bb17d35e73"
    }
}
